src/typecode/magic2.py

```diff
@@ -79,18 +79,60 @@
 TYPECODE_LIBMAGIC_DATABASE = 'typecode.libmagic.db'
 
 
+def load_lib_failover():
+  # loader from python-magic
+  # License: MIT 
+  # Copyright: Adam Hupp
+  libmagic = None
+  # Let's try to find magic or magic1
+  dll = ctypes.util.find_library('magic') \
+        or ctypes.util.find_library('magic1') \
+        or ctypes.util.find_library('cygmagic-1') \
+        or ctypes.util.find_library('libmagic-1') \
+        or ctypes.util.find_library('msys-magic-1')  # for MSYS2
+
+  # necessary because find_library returns None if it doesn't find the library
+  if dll:
+      libmagic = ctypes.CDLL(dll)
+
+  if not libmagic or not libmagic._name:
+      windows_dlls = ['magic1.dll', 'cygmagic-1.dll', 'libmagic-1.dll', 'msys-magic-1.dll']
+      platform_to_lib = {'darwin': ['/opt/local/lib/libmagic.dylib',
+                                    '/usr/local/lib/libmagic.dylib'] +
+                                  # Assumes there will only be one version installed
+                                  glob.glob('/usr/local/Cellar/libmagic/*/lib/libmagic.dylib'),  # flake8:noqa
+                        'win32': windows_dlls,
+                        'cygwin': windows_dlls,
+                        'linux': ['libmagic.so.1'],
+                        # fallback for some Linuxes (e.g. Alpine) where library search does not work # flake8:noqa
+                        }
+      platform = 'linux' if sys.platform.startswith('linux') else sys.platform
+      for dll in platform_to_lib.get(platform, []):
+          try:
+              libmagic = ctypes.CDLL(dll)
+              break
+          except OSError:
+              pass
+
+  if not libmagic or not libmagic._name:
+      return None
+  return libmagic
```

Comment:
Do you know if there could be a way to test/check the version of this libmagic? Because unless this is close enough to the one provided as a plugin, there could be some magic database inconsistencies.

Reply:
We could do a check, but we don't really know what threshold we should really put there. How would you define close enough?
I would suggest to leave a warning, and let the community report any issue.

We could have a TAINTED flag like linux does which could be put in the typecode dependent reports

Reply:
Well in hindsight, let's leave it this way.... after all in https://github.com/nexB/scancode-plugins/blob/main/builtins/typecode_libmagic_system_provided/src/typecode_libmagic/__init__.py I do not even check for version